gst/edge/edge_sink.c

```diff
@@ -0,0 +1,344 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2022 Samsung Electronics Co., Ltd.
+ *
+ * @file    edge_sink.h
+ * @date    01 Aug 2022
+ * @brief   Publish incoming streams
+ * @author  Yechan Choi <yechan9.choi@samsung.com>
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @bug     No known bugs
+ *
+ */
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "edge_sink.h"
+
+GST_DEBUG_CATEGORY_STATIC (gst_edgesink_debug);
+#define GST_CAT_DEFAULT gst_edgesink_debug
+
+/**
+ * @brief the capabilities of the inputs.
+ */
+static GstStaticPadTemplate sinktemplate = GST_STATIC_PAD_TEMPLATE ("sink",
+    GST_PAD_SINK,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS_ANY);
+
+/**
+ * @brief edgesink properties
+ */
+enum
+{
+  PROP_0,
+
+  PROP_HOST,
+  PROP_PORT,
+
+  PROP_LAST
+};
+
+#define gst_edgesink_parent_class parent_class
+G_DEFINE_TYPE (GstEdgeSink, gst_edgesink, GST_TYPE_BASE_SINK);
+
+static void gst_edgesink_set_property (GObject * object,
+    guint prop_id, const GValue * value, GParamSpec * pspec);
+
+static void gst_edgesink_get_property (GObject * object,
+    guint prop_id, GValue * value, GParamSpec * pspec);
+
+static void gst_edgesink_finalize (GObject * object);
+
+static gboolean gst_edgesink_start (GstBaseSink * basesink);
+static GstFlowReturn gst_edgesink_render (GstBaseSink * basesink,
+    GstBuffer * buffer);
+static gboolean gst_edgesink_set_caps (GstBaseSink * basesink, GstCaps * caps);
+
+static gchar *gst_edgesink_get_host (GstEdgeSink * self);
+static void gst_edgesink_set_host (GstEdgeSink * self, const gchar * host);
+
+static guint16 gst_edgesink_get_port (GstEdgeSink * self);
+static void gst_edgesink_set_port (GstEdgeSink * self, const guint16 port);
+
+/**
+ * @brief initialize the class
+ */
+static void
+gst_edgesink_class_init (GstEdgeSinkClass * klass)
+{
+  GObjectClass *gobject_class;
+  GstElementClass *gstelement_class;
+  GstBaseSinkClass *gstbasesink_class;
+
+  gstbasesink_class = (GstBaseSinkClass *) klass;
+  gstelement_class = (GstElementClass *) gstbasesink_class;
+  gobject_class = (GObjectClass *) gstelement_class;
+
+  gobject_class->set_property = gst_edgesink_set_property;
+  gobject_class->get_property = gst_edgesink_get_property;
+  gobject_class->finalize = gst_edgesink_finalize;
+
+  g_object_class_install_property (gobject_class, PROP_HOST,
+      g_param_spec_string ("host", "Host", "The hostname to send", DEFAULT_HOST,
```

Comment:
I think it's self hostname to accept the connection from edgesrc.
```suggestion
      g_param_spec_string ("host", "Host", "The hostname to listen", DEFAULT_HOST,
```